rir/src/interpreter/RirDataWrapper.h

```diff
@@ -13,32 +13,40 @@
 
 namespace rir {
 
+#define RIR_DATA_WRAPPER_MAGIC 0xda7a0403
+
 struct data_header {
-    // magic number to differentiate RIR objects
-    // needs to be in sync with gnur context.c
-    uint32_t magic;
+    // magic number to differentiate RIR objects needs sync with gnur context.c
+    const uint32_t magic;
+
+    // magic number to differentiate different RIR data wrappers
+    const uint32_t wrapper;
+
+    // #SEXPs stored in an vector* at the begging of the struct to be preserved
+    const uint32_t gc;
 };
 
 template <typename BASE, uint32_t MAGIC>
 struct RirDataWrapper {
```

Comment:
It is a kind of cryptic struct strongly connected on how gnu-r uses it. So maybe some comments are worth. But it is not true that the gc is a subclass responsibility. Every `RirDataWrapper` needs to store how many objects that need to be traced by the gc it stores. Then each subclass only implements a function `gcData` that return a pointer to an array of the corresponding size so that gnu-r can walk through all the data. 

Then, two magic numbers is just a matter of is to simplify gnu-r checks. So, gnu-r treats all wrappers that same way, they only need to know the gc objects to trace. I do not want to have tests full of or for all the eventual wrappers in the gnu-r gc. Then, at materialization time we need to disambiguate into different structures. And they are in a inner struct and not as just struct const fields because again, I inspired in `RirRuntimeObjects`